### DISCUSSION TITLE
FULL EXAMPLE READING DATA

### DISCUSSION THREAD

**Discussion by lalo4ka** [rm5248/JavaSerial#34] [General]
Hi,  could you upload a complete example of how to read data?

**Comment by rm5248** [comment]
Something like this should work:

```
SerialPort s;
 try {
    s = new SerialPort( "/dev/ttyUSB0" );
} catch (NoSuchPortException e) {
    System.err.println( "Oh no!  That port doesn't exist!" );
} catch (NotASerialPortException e) {
    System.err.println( "Oh no!  That's not a serial port!" );
}

InputStream is = s.getInputStream();

System.out.println("Read byte: " + is.read());
```

The `InputStream` acts exactly the same as any other `InputStream` that exists.  This tutorial has examples of how to use `InputStream`s: https://jenkov.com/tutorials/java-io/inputstream.html

**Reply by lalo4ka** [reply]
Thank you very much, my code now works.

My project is to read the data from a scale (Rhino Brand), however first you must send a command to request the data. and this was the step I was missing.

I share my code in case anyone needs it:

`SerialPort s = null;
        try {
            s = new SerialPort("/dev/ttyUSB0");
        } catch (NoSuchPortException e) {
            System.err.println("Oh no!  That port doesn't exist!");
        } catch (NotASerialPortException e) {
            System.err.println("Oh no!  That's not a serial port!");
        }

        OutputStream os = s.getOutputStream();
        // Send Letter P or number 80 en hexidecimal
        os.write("P".getBytes());

        InputStream is = s.getInputStream();        

        BufferedReader br = new BufferedReader(new InputStreamReader(is));
        System.out.println(br.readLine());`
